src/Orleans/Providers/IMemoryStorageGrain.cs

```diff
@@ -23,13 +21,9 @@ public interface IMemoryStorageGrain : IGrainWithIntegerKey
         /// <param name="stateStore">The name of the store that is used to store this grain state</param>
         /// <param name="grainStoreKey">Store key for this grain.</param>
         /// <param name="grainState">New state data to be stored for this grain.</param>
-        /// <param name="eTag">The previous etag that was read.</param>
-        /// <returns>Value promise of the etag of the update operation for stored grain state for the specified grain.</returns>
-        Task<string> WriteStateAsync(string stateStore, string grainStoreKey, IDictionary<string, object> grainState, string eTag);
-
-        /// <summary>
-        /// Async method to cause deletion of the specified grain state data from memory store.
-        /// </summary>
+        /// <returns>Completion promise with new eTag for the update operation for stored grain state for the specified grain.</returns>
+        Task<string> WriteStateAsync(string grainType, string grainId, IGrainState grainState);
```

Comment:
Should this be returning the `ETag`? - `ETag`s aren't a concept which all storage systems have.